contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java

```diff
@@ -834,17 +840,16 @@ private boolean isPartitionRequired(int opid, List<KafkaConsumer.KafkaMeterStats
         return false;
       }
 
+      if (!isPartitionBasedOnLoad) {
```

Comment:
I do think the logic is not correct in the if block from line 599. The new kafka partition should be merged into existing partition